flake.nix

```diff
@@ -10,13 +10,27 @@
   outputs = { self, nixpkgs, nixlib }:
 
   # Library modules (depend on nixlib)
-  {
+  rec {
     # export all generator formats in ./formats
     nixosModules = nixlib.lib.mapAttrs' (file: _: {
       name = nixlib.lib.removeSuffix ".nix" file;
       # The exported module should include the internal format* options
       value.imports = [ (./formats + "/${file}") ./format-module.nix ];
+      # !!! TODO: is there a better way to extract formatAttr for the format?
```

Comment:
yes.

check ./format-module.nix, it makes "formatAttr" an option

that means you can use config.formatAttr instead

```
let
  system = nixpkgs.lib.nixosSystem { ... };
in
  system.config.system.build.${system.config.formatAttr}
```

Reply:
Thanks for pointing out the options interface to `nixos-generators` I had glossed over that.  Take a look at my latest commit.